driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -572,26 +615,7 @@ public void onSet(Connection connection, Message.Response response, long latency
                         if (retry == null)
```

Comment:
That case is for an unexpected error code; it will cause the error to be reported directly to the user (`DefaultResultSetFuture.onSet`), which sounds reasonable.